mayavi/sources/vtk_file_reader.py

```diff
@@ -75,18 +76,20 @@ def _file_path_changed(self, fpath):
                 n = self.reader.number_of_outputs
             except AttributeError: # for VTK >= 4.5
                 n = self.reader.number_of_output_ports
+
             outputs = []
             for i in range(n):
                 outputs.append(self.reader.get_output(i))
-            self.outputs = outputs
 
-            # FIXME: Only the first output goes through the assign
-            # attribute filter.
-            aa = self._assign_attribute
-            self.configure_input_data(aa, outputs[0])
-            self.update_data()
-            aa.update()
-            outputs[0] = aa.output
+            # FIXME: currently handling only one output (the first one) with
+            # assign attributes.
+            for index, output in enumerate(outputs):
```

Comment:
It only has one, but then probably it's just missing a break, or I should simply remove the for loop altogether. The critical point is the has_attributes part. If you say it's a requirement that we only have one output, then it's certainly the case. I revise it.